sdk/internal/testframework/README.md

```diff
@@ -0,0 +1,161 @@
+var clientsMap map[string]*testContext = make(map[string]*testContext)
+
+// recordedTestSetup is called before each test execution by the test suite's BeforeTest method
+func recordedTestSetup(t *testing.T, testName string, mode testframework.RecordMode) {
+    var accountName string
+    var suffix string
+    var cred *SharedKeyCredential
+    var secret string
+    var uri string
+    assert := assert.New(t)
+
+    // init the test framework
+    context := testframework.NewTestContext(func(msg string) { assert.FailNow(msg) }, func(msg string) { t.Log(msg) }, func() string { return testName })
+    //mode should be testframework.Playback. This will automatically record if no test recording is available and playback if it is.
+    recording, err := testframework.NewRecording(context, mode) 
+    assert.Nil(err)
+```
+
+After creating the TestContext, it must be passed to a new instance of `Recording` along with the current test mode.
+`Recording` is the main component of the testframework package.
+
+```go
+//func recordedTestSetup(t *testing.T, testName string, mode testframework.RecordMode) {
+//  <...>
+    recording, err := testframework.NewRecording(context, mode)
+    assert.Nil(err)
+```
+
+### Initializing recorded variables
+
+A key component to recorded tests is recorded variables.
+They allow creation of values that stay with the test recording so that playback of service operations is consistent.
+
+In the snippet below we are calling `GetRecordedVariable` to acquire details such as the service account name and
+client secret to configure the client.
+
+```go
+//func recordedTestSetup(t *testing.T, testName string, mode testframework.RecordMode) {
+//  <...>
+    accountName, err := recording.GetRecordedVariable(storageAccountNameEnvVar, testframework.Default)
+    suffix := recording.GetOptionalRecordedVariable(storageEndpointSuffixEnvVar, DefaultStorageSuffix, testframework.Default)
+    secret, err := recording.GetRecordedVariable(storageAccountKeyEnvVar, testframework.Secret_Base64String)
+    cred, _ := NewSharedKeyCredential(accountName, secret)
+    uri := storageURI(accountName, suffix)
+```
+
+The last step is to instrument your client by replacing its transport with your `Recording` instance.
+`Recording` satisfies the `azcore.Transport` interface.
+
+```go
+//func recordedTestSetup(t *testing.T, testName string, mode testframework.RecordMode) {
+//  <...>
+    // Set our client's HTTPClient to our recording instance.
+    // Optionally, we can also configure MaxRetries to -1 to avoid the default retry behavior.
+    client, err := NewTableServiceClient(uri, cred, &TableClientOptions{HTTPClient: recording, Retry: azcore.RetryOptions{MaxRetries: -1}})
+    assert.Nil(err)
+
+    // either return your client instance, or store it somewhere that your test can use it for test execution.
+    clientsMap[testName] = &testContext{client: client, recording: recording, context: &context}
```

Comment:
It doesn't seem same as the one provided in `testcontext.go`. Any implementing test suite is required to create this one on their own?

```go
type testContext struct {
    client     *URL
    recording  *testframework.Recording
    context    *testframework.TestContext
}
```

Reply:
This is just a helper for keeping track of test state. Different tests may have different requirements as to what they want to store as state. However, I agree the name makes it confusing with the testframeowk.TestContext type.